  <launch>
    <arg name="map_size_x" value="42.0"/>
    <arg name="map_size_y" value="40.0"/>
    <arg name="map_size_z" value=" 5.0"/>

    <arg name="odom_topic" value="visual_slam/odom" />

    <!-- random map -->
    <node pkg ="map_generator" name ="random_forest" type ="random_forest" output = "screen">    
        <param name="map/x_size"     value="36" />
        <param name="map/y_size"     value="20" />
        <param name="map/z_size"     value="3" />
        <param name="map/resolution" value="0.1"/>        

        <param name="ObstacleShape/seed" value="1"/>
        
        <param name="map/obs_num"    value="55"/>
        <param name="ObstacleShape/lower_rad" value="1.0"/>
        <param name="ObstacleShape/upper_rad" value="1.0"/>
        <param name="ObstacleShape/lower_hei" value="0.5"/>
        <param name="ObstacleShape/upper_hei" value="3.0"/>        

        <param name="map/circle_num" value="20"/>        
        <param name="ObstacleShape/radius_l" value="0.7"/>        
        <param name="ObstacleShape/radius_h" value="0.5"/>        
        <param name="ObstacleShape/z_l" value="0.7"/>        
        <param name="ObstacleShape/z_h" value="0.8"/>        
        <param name="ObstacleShape/theta" value="0.5"/>        

        <param name="sensing/radius" value="5.0"/>        
        <param name="sensing/rate"   value="1.0"/>  
        
        <param name="min_distance" value="1.2"/>         
        <remap from="/map_generator/global_cloud" to="/global_map"/>

    </node>


    <!-- manual map -->
    <!-- <node pkg ="map_generator" name ="map_generator_easy" type ="map_generator_easy.py" output = "screen" respawn="true" >
        <param name="map/x_size"     value="20" />
        <param name="map/y_size"     value="10" />
        <param name="map/z_size"     value="3" />
        <param name="map/resolution" value="0.1"/>

        <param name="map/add_floor" value="false"/>
        <param name="map/add_ceiling" value="false"/>

        <param name="sensing/rate" value="1.0"/>
        <remap from="/map_generator/global_cloud" to="/global_map"/>

    </node> -->


    <include file="$(find ego_planner)/launch/run_in_sim_tracker.launch">
        <arg name="drone_id" value="0"/>

        <arg name="init_x" value="-17.0"/>
        <arg name="init_y" value="0.0"/>
         <arg name="init_z" value="1.0"/>

        <arg name="target_x" value="10.0"/>
        <arg name="target_y" value="-3.5"/>
        
        <arg name="target_z" value="1.0"/>

        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
        <arg name="odom_topic" value="$(arg odom_topic)"/>
    </include>

    <include file="$(find ego_planner)/launch/run_in_sim_target.launch">
        <arg name="drone_id" value="1"/>

        <arg name="init_x" value="-15.0"/>
        <arg name="init_y" value="0.0"/>
         <arg name="init_z" value="1.0"/>

        <arg name="target_x" value="10.0"/>
        <arg name="target_y" value="-3.5"/>
        <arg name="target_z" value="1.0"/>

        <arg name="map_size_x" value="$(arg map_size_x)"/>
        <arg name="map_size_y" value="$(arg map_size_y)"/>
        <arg name="map_size_z" value="$(arg map_size_z)"/>
        <arg name="odom_topic" value="$(arg odom_topic)"/>
    </include>

  </launch>
